### DISCUSSION TITLE
Enhanced Kotlin Inspections

### DISCUSSION THREAD

**Discussion by zilinjak** [JetBrains/Qodana#187] [Q&A]
Hello,
I have recently been exploring the use of Qodana and Kotlin inspections to identify various issues such as:
- Error handling
- Initialization
- Performance
- Resource management
- Security
- Threading issues

While using IntelliJ IDEA to check what inspections Java supports, I've noticed that I can analyze all the categories mentioned above. 
However, when it comes to inspecting Kotlin code, I've found that only a limited number of inspections are available. 
The majority of the categories I mentioned earlier seem to be missing. The only category that seems to work for me is the "JVM Languages" category, which includes features like detecting "Possibly blocking call in non-blocking context." Unfortunately, many important categories cannot be properly analyzed in Kotlin.

If you have any insights or solutions regarding this issue, I would greatly appreciate your guidance.

Thank you.

**Comment by brichbash** [comment]
Hello,

> The only category that seems to work for me is the "JVM Languages"

Do you mean that you have some issues when using inspections from "Kotlin" category or that you they don't meet your needs?

Qodana supports all inspections that are provided by [Kotlin Plugin](https://plugins.jetbrains.com/plugin/6954-kotlin), which is bundled with IntelliJ IDEA by default. I understand that in some areas Kotlin inspections may be underrepresented when compared to Java. However, the Kotlin team is actively working on them, and you can help them by creating/voting for existing feature requests [on their tracker](https://youtrack.jetbrains.com/issues?q=in:%20KTIJ).

**Reply by zilinjak** [reply]
I believe there is room for improvement in inspections, particularly in areas such as `Performance`, `Thread issues`, and `Security`.

If I was not wrong with my question and solution, that there is no way how we can inspect categories above. 
This Q can be closed and I will create some requests [on their tracker](https://youtrack.jetbrains.com/issues?q=in:%20KTIJ).

**Reply by brichbash** [reply]
No, however, there are issues on the tracker for some of them: [KTIJ-9743](https://youtrack.jetbrains.com/issue/KTIJ-9743), [KTIJ-22609](https://youtrack.jetbrains.com/issue/KTIJ-22609). You can vote for them to get notified of any updates and also help the team in prioritizing the issues.
